va/drm/va_drm_utils.c

```diff
@@ -110,6 +112,14 @@ VA_DRM_GetDriverName(VADriverContextP ctx, char **driver_name_ptr, int candidate
         return VA_STATUS_ERROR_INVALID_DISPLAY;
 
     *driver_name_ptr = va_DRM_GetDrmDriverName(drm_state->fd);
+
+    /* Map vgem to WSL2 for Windows subsystem for linux */
+    struct utsname sysinfo = {};
+    if (!strncmp(*driver_name_ptr, "vgem", 4)
+        && !uname(&sysinfo)
+        && strstr(sysinfo.release, "WSL"))
+        *driver_name_ptr = strdup("WSL");
```

Comment:
You are losing memory here since driver_name_ptr containerd "vgem" which needs to be freed:
```
if (...) {
  free(*driver_name_ptr);
  *driver_name_ptr = strdup("WSL");
}
```

Reply:
Great catch, fixed it